### PR TITLE
feat:JRD framework setup & camel spring transaction(RDCC-975)

### PR DESCRIPTION
**Pull request by abhijitdiwan**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-975

### Change description ###
feat:JRD framework setup & camel spring transaction(RDCC-975)

**Does this PR introduce a breaking change?** (check one with "x")
No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/processor/JudicialUserProfileProcessor.java [hunk 1]
```diff
@@ -15,11 +15,21 @@
...
+        List<JudicialUserProfile> judicialUserProfiles;
+
+        if (exchange.getIn().getBody() instanceof List) {
+            judicialUserProfiles = (List<JudicialUserProfile>) exchange.getIn().getBody();
+        } else {
+            JudicialUserProfile judicialUserProfile = (JudicialUserProfile) exchange.getIn().getBody();
+            judicialUserProfiles = new ArrayList<>();
+            judicialUserProfiles.add(judicialUserProfile);
+        }
+
```
[the remark this answers is not in the document]
Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/juddata/camel/processor/JudicialUserProfileProcessor.java]: yes

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/route/ParentOrchestrationRoute.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    SpringTransactionPolicy springTransactionPolicy;
+
+    @SuppressWarnings("unchecked")
+    @Transactional
+    public void startRoute() throws Exception {
+
+        CamelContext camelContext = ctx.getBean(CamelContext.class);
+        String parentRouteName = camelContext.getGlobalOptions().get(PARENT_ROUTE_NAME);
+        String parentName = ROUTE + "." + parentRouteName;
+        String childNames = ROUTE + "." + parentRouteName + "." + CHILD_ROUTE_NAME;
```
[the remark this answers is not in the document]
Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/juddata/camel/route/ParentOrchestrationRoute.java]: yes...We have on class child route using this. We will remove these and that child route

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/route/ParentOrchestrationRoute.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+        List<String> childes = environment.containsProperty(childNames)
```
[the remark this answers is not in the document]
Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/juddata/camel/route/ParentOrchestrationRoute.java]: ok

File: src/main/resources/application-jrd-router.yaml [hunk 4]
```diff
@@ -19,21 +21,31 @@ parent-route:
...
     mapper-class: JudicialOfficeAuthorisationRowMapper
     csv-binder-object: JudicialOfficeAuthorisation
 
-
-
+  judicial-office-appointment:
+    id: judicial-office-appointment
+    truncate-sql:
+      sql:truncate judicial_office_appointment restart identity?dataSource=dataSource
+    sql:
```
[the remark this answers is not in the document]
Comment (abhijitdiwan) [reply src/main/resources/application-jrd-router.yaml]: it is truncate...used for truncation specific

[the remark this answers is not in the document]
Comment (shashank-rastogi) [reply src/main/resources/application-jrd-router.yaml]: i was talking abt sql

[the remark this answers is not in the document]
Comment (abhijitdiwan) [reply src/main/resources/application-jrd-router.yaml]: updated

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/util/MappingConstants.java [hunk 5]
```diff
@@ -1,11 +1,16 @@
 package uk.gov.hmcts.reform.juddata.camel.util;
 
+import java.sql.Timestamp;
+import java.util.Date;
+
 public interface MappingConstants {
 
     String PARENT_ROUTE_NAME = "parent-route";
 
     String CHILD_ROUTE_NAME = "child-route";
```
[the remark this answers is not in the document]
Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/juddata/camel/util/MappingConstants.java]: updated names. But those needed

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/processor/ExceptionProcessor.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+public class ExceptionProcessor implements Processor {
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        Exception exception = (Exception) exchange.getProperty(EXCEPTION_CAUGHT);
+        log.error("::::exception in route for data processing::::" + exception);
```
Comment (shashank-rastogi) [inline src/main/java/uk/gov/hmcts/reform/juddata/camel/processor/ExceptionProcessor.java]: can you add the route name  in the exception message?

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/processor/FileReadProcessor.java [hunk 7]
```diff
@@ -19,7 +19,7 @@ public void process(Exchange exchange) {
         String blobFilePath = (String) exchange.getProperty(BLOBPATH);
         CamelContext context = exchange.getContext();
         ConsumerTemplate consumer = context.createConsumerTemplate();
-        exchange.getMessage().setBody(consumer.receiveBody(blobFilePath, 600000));
+        exchange.getMessage().setBody(consumer.receiveBody(blobFilePath, 180000));
```
Comment (shashank-rastogi) [inline src/main/java/uk/gov/hmcts/reform/juddata/camel/processor/FileReadProcessor.java]: lets reduce it back to 600000

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/route/ParentOrchestrationRoute.java [hunk 8]
```diff
@@ -77,11 +77,11 @@ public void startRoute() throws Exception {
         String parentName = ROUTE + "." + parentRouteName;
         String childNames = ROUTE + "." + parentRouteName + "." + CHILD_ROUTES;
 
-        List<String> childrenList = environment.containsProperty(childNames)
+        List<String> dependents = environment.containsProperty(childNames)
```
Comment (shashank-rastogi) [inline src/main/java/uk/gov/hmcts/reform/juddata/camel/route/ParentOrchestrationRoute.java]: dependant spelling is wrong.. and i think children is still a better choice of word than dependant or if you want to use dependant than perhaps better choice is "dependantRoutes"

Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/juddata/camel/route/ParentOrchestrationRoute.java]: done

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/processor/FileReadProcessor.java (hunk 7 above)
Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/juddata/camel/processor/FileReadProcessor.java]: done

**Pull request merged**